radio/src/gui/colorlcd/model_mixer_scripts.cpp

```diff
@@ -246,20 +259,22 @@ void ModelMixerScriptsPage::build(FormWindow * window, int8_t focusIdx)
     }
     
     // LUAx label
-    auto txt = new StaticText(window, grid.getLabelSlot(),
+    auto const txt = new StaticText(window, grid.getLabelSlot(),
                               std::string("LUA") + std::to_string(idx + 1),
                               BUTTON_BACKGROUND, COLOR_THEME_PRIMARY1 | CENTERED);
     
-    Button* button = new ScriptLineButton(window, grid.getFieldSlot(), sd, runtimeData);
+    Button* const button = new ScriptLineButton(window, grid.getFieldSlot(), sd, runtimeData);
 
+    ScriptData* const sp = &sd;
     button->setPressHandler([=]() -> uint8_t {
-      Menu* menu = new Menu(window);
+      Menu* const menu = new Menu(window);
       menu->addLine(STR_EDIT, [=]() { editLine(window, idx); });
 
       if (runtimeData != nullptr) {
-        menu->addLine(STR_RESET, [=]() {
-          memset((void*)&sd, 0, sizeof(sd));
-          // TODO: anything else? reload scripts???
+        menu->addLine(STR_DELETE, [=]() {
+          memset((void*)sp, 0, sizeof(sd));
+          // scripts input/output should be cleared
+          // can't be done here, because script may be scheduled again (see: ***)
```

Comment:
Reloading the scripts could still be triggered here, right?

Did you try with this?
```
LUA_LOAD_MODEL_SCRIPTS();
```

Reply:
See comment in the code.
It needs an event when the script is successfully loaded and run for the first time to define the inputs and outputs. Before that, the data-structures do not resemble the situation the scripts wants to define.

Reply:
@jfrickmann Perhaps you have an idea how to achieve this?

Reply:
> @jfrickmann Perhaps you have an idea how to achieve this?
`LUA_LOAD_MODEL_SCRIPTS()` sets the Lua task state to `INTERPRETER_RELOAD_PERMANENT_SCRIPTS`:
https://github.com/EdgeTX/edgetx/blob/2e5dbce2cbc71da66c4e484cf6d82a3d44979c5c/radio/src/lua/lua_api.h#L179-L180
When the Lua task has loaded all of the scripts, then `luaState == INTERPRETER_RUNNING` - so you could wait for that before re-reading the data structures.

Reply:
My hope that there is an gui-event-queue, where one could enqueue an vent from the lua-task to the gui-task?

If that is not possible: are the `rebuild_()` element-functions called periodically (assume, that not)?
Is the event-handler `checkEvents()` called periodically (or can I achieve that)?

Reply:
> My hope was that there is an gui-event-queue where one could enqueue an event from the lua-task to the gui-task?

No such thing.

> If that is not possible: are the `rebuild_()` element-functions called periodically (assume, that not)? Is the event-handler `checkEvents()` called periodically (or how can I achieve that an event gets routed to this handler)?

Yes, checkEvents() is called in every GUI cycle.

Reply:
Many thanks for that hint, @jfrickmann. Now got if fully working.